### PR TITLE
Add a TypeScript step to `create-astro`

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

This PR adds a step to `create-astro` to allow users to configure how strict they want their TypeScript install to be

image

I've set the "Strict" preset as the recommended one ~~in hope of making everyone's code better~~ as it's the one TypeScript itself recommend for new projects (same goes for all our other recommended settings)

The "Stricter" settings were mostly sourced from `create-t3-app`, it has very cool settings for TypeScript

## Testing

Tested manually

## Docs

Will PR docs if needed

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
Love this!

File: packages/create-astro/src/index.ts [hunk 1]
```diff
@@ -284,10 +285,61 @@ export async function main() {
...
+	const tsResponse = await prompts(
+		{
+			type: 'select',
+			name: 'typescript',
+			message:
+				'Astro is a TypeScript-first framework. How strict would you like the typechecking to be?',
+			choices: [
+				{
+					title: 'Relaxed',
+					description: "Not a big fan of TypeScript? No issues, we'll keep this kind",
```
Comment (sarah11918) [inline packages/create-astro/src/index.ts:302]:
```suggestion
					description: "Not a big fan of TypeScript? No worries, we'll keep things casual",
```
I LOVE this option! This is such user empathy!  😄 

I would suggest adjusting the message a little bit, though. When I read it, I read "kind" expecting "kind of..." So, I thought it was going to say, "We'll keep this kind of...." and then it stopped.  

Often designing messages for the widest audience possible, we would try to avoid slang etc. So, you can't go wrong with a very factual description here of what will/won't happen. But, I think if you want to convey this friendliness, there are still some choices that could work.  Maybe like....

First part:  I think "No worries" is much more common across an international audience than "No issues"  (issues might make people think of bugs? or is there an "issue" with this process?) I'd consider "worries" instead.

Second part: this is a little harder, but here are some possibilities that might be a bit more accessible while still being fun, maybe?

- we'll keep things casual
- we'll take it easy on you
- we won't make you use it / you don't have to use it
- we'll turn off the red squigglies
- we'll try to help, but won't insist
... ?

Comment (Princesseuh) [reply packages/create-astro/src/index.ts:302]: I agree! I used `No issues` because `No worries` is used earlier in the steps and my brain immediately went in "Find synonyms" mode 😄

For the second part, we can't say that the user won't have to use it or that it'll disable anything because it effectively doesn't. TypeScript is always enabled, just with friendlier settings. `we'll keep things casual` seems good to me!

**Comment by FredKSchott** [comment]
With @Princesseuh's blessing I just took a pass on the prompts, here's where I landed:

## Before

Screen Shot 2022-08-05 at 3 08 46 PM

## After

Screen Shot 2022-08-05 at 3 38 30 PM

Screen Shot 2022-08-05 at 3 35 07 PM

File: packages/create-astro/src/index.ts (hunk 1 above)
Comment (FredKSchott) [reply packages/create-astro/src/index.ts]: Here were my suggestions, lmk what you think! https://github.com/withastro/astro/pull/4179#issuecomment-1207069870

Comment (FredKSchott) [reply packages/create-astro/src/index.ts]: One thought on formatting: because this is the last question, what if the "don't love typescript" message came after "setup complete!" That way it wouldn't break up the series of prompts mid-way through.

**Comment by Princesseuh** [comment]
I feel like giving the option "I don't want to use TS" and the text are selling something that doesn't exist to users. 

It's not just that Astro support TypeScript, it's TypeScript, even on the most relaxed settings, there will be TypeScript errors, even without a tsconfig.json etc

**Comment by FredKSchott** [comment]
So I was trying to call that out here, basically saying "we understand that you prefer not to, but Astro needs to, so trust us and don't worry about it". Could this be rewritten to be more clear?

Let me reword this to be more of a warning, since it sounds like the "Astro needs to" part of it was lost. Is this any better?

Screen Shot 2022-08-05 at 9 03 42 PM

**Review by FredKSchott — APPROVED** [review]
LGTM! Merging to get this out in the next release but I'm happy to keep workshopping the wording in a future PR.
